File: Ejercicios/C#/Clases Abstractas (o equivalente)/Vehiculo_Moto.cs
using System;

public abstract class Vehiculo
{
    public string Placa;

    public Vehiculo()
    {
        Placa = "XXX000";
    }

    public Vehiculo(string placa)
    {
        Placa = placa;
    }

    public Vehiculo(string placa, string pais)
    {
        Placa = placa + "-" + pais;
    }

    public Vehiculo(Vehiculo otro)
    {
        Placa = otro.Placa;
    }

    public abstract void Mover();
}

public class Moto : Vehiculo
{
    public Moto() : base() { }

    public Moto(string placa) : base(placa) { }

    public Moto(string placa, string pais) : base(placa, pais) { }

    public Moto(Moto otra) : base(otra) { }

    public override void Mover() => Console.WriteLine("La moto se mueve");
}

File: Ejercicios/GO/Clases Abstractas (o equivalente)/Vehiculo_Moto.go
package main

type Vehiculo interface {
	Mover() string
}

type Moto struct {
	Placa string
}

func NewMoto() *Moto {
	return &Moto{Placa: "XXX000"}
}

func NewMotoConPlaca(placa string) *Moto {
	return &Moto{Placa: placa}
}

func NewMotoPlacaPais(placa, pais string) *Moto {
	return &Moto{Placa: placa + "-" + pais}
}

func NewMotoCopy(m *Moto) *Moto {
	return &Moto{Placa: m.Placa}
}

func (m *Moto) Mover() string {
	return "La moto se mueve"
}
